Crash on Reviewer start - NullPointerException restoring preferences

###### Reproduction Steps

1. Open AnkiDroid to the Reviewer directly (e.g., via a notification or widget)
2. 
3. 


###### Expected Result

Reviewer starts on current card for the deck and you proceed

###### Actual Result

Crash, seemingly because onCreate() in this pathway touches the Collection during restorePreferences() (since some preferences are stored in the Collection.conf), but collection loading is async and onCollectionLoaded() hasn't been called yet

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/9964d8fb-0eab-4d4a-a0f6-403350bbbf60

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.Reviewer}: java.lang.NullPointerException: Attempt to invoke virtual method 'org.json.JSONObject com.ichi2.libanki.Collection.getConf()' on a null object reference
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2632)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2698)
at android.app.ActivityThread.access$1000(ActivityThread.java:195)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533)
at android.os.Handler.dispatchMessage(Handler.java:111)
at android.os.Looper.loop(Looper.java:207)
at android.app.ActivityThread.main(ActivityThread.java:5856)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1026)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:887)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.json.JSONObject com.ichi2.libanki.Collection.getConf()' on a null object reference
at com.ichi2.anki.AbstractFlashcardViewer.restorePreferences(AbstractFlashcardViewer.java:1803)
at com.ichi2.anki.Reviewer.restorePreferences(Reviewer.java:397)
at com.ichi2.anki.AbstractFlashcardViewer.onCreate(AbstractFlashcardViewer.java:856)
at android.app.Activity.performCreate(Activity.java:6285)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2585)
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java
@@ -849,6 +849,8 @@ public abstract class AbstractFlashcardViewer extends NavigationDrawerActivity {
 
         registerExternalStorageListener();
 
+        restoreCollectionPreferences();
+
         initLayout();
 
         setTitle();
@@ -1746,6 +1748,12 @@ public abstract class AbstractFlashcardViewer extends NavigationDrawerActivity {
             this.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
         }
 
+        return preferences;
+    }
+
+
+    private void restoreCollectionPreferences() {
+
         // These are preferences we pull out of the collection instead of SharedPreferences
         try {
             mShowNextReviewTime = getCol().getConf().getBoolean("estTimes");
@@ -1753,8 +1761,6 @@ public abstract class AbstractFlashcardViewer extends NavigationDrawerActivity {
         } catch (JSONException e) {
             throw new RuntimeException();
         }
-
-        return preferences;
     }
 
 

